docs/specs/future-of-workload.md

```diff
@@ -0,0 +1,43 @@
+  
+## Problem statement
+
+The current design of the workload has led to some usability issues, and after a lot of deep analysis, we finally understand what is the root cause of these issues. SDK Workloads are meant to be used for tooling components, which lifetime should be tied to the lifetime of the SDK itself. When we were originally designing the workload, we believed that both DCP and Dashboard were tooling components, and that they should be installed as part of the workload. However, we have come to realize that this is not the case. DCP and Dashboard are tightly coupled to the version of Aspire.Hosting package that the AppHost project references, and therefore are part of the AppHost itself, as opposed to tooling that supports it. This is the root cause of the issues that we have seen with the workload, such as the need to ensure that the version of the workload installed matches the project reference of Aspire.Hosting, which is also a problem when trying to work with multiple Aspire projects that reference different versions of Aspire.Hosting. SDK typically makes the assumption that customers want to always be in the latest version of all of the workloads, which is likely the right thing to do for tooling components; however, this is a problem for .NET Aspire given the tight coupling between the AppHost and the DCP/Dashboard.
+
+## Proposed solution
+
+Given the above, we propose to move the DCP and Dashboard components out of the workload, and into NuGet packages that are referenced as dependencies of the Aspire.Hosting.AppHost package. This will allow the AppHost project to reference the version of Aspire.Hosting that it wants, and then the DCP and Dashboard will be installed as dependencies of that version of Aspire.Hosting. This will allow the AppHost project to have full control over the version of the DCP and Dashboard that it wants to use, and will also allow customers to have multiple AppHost projects that reference different versions of Aspire.Hosting without any issues. This will also allow us to update the DCP and Dashboard independently of the workload, which will allow us to ship updates to these components more frequently and with less risk, with a far less complex compatibility matrix.
+
+Also, there is no real need of including the Aspire.Hosting package as part of the workload, as there is no guarantee that this is the version that the AppHost project will want to use, therefore we would also remove this package from the workload. In the end, the workload will only contain the Aspire.Hosting.SDK and Aspire.ProjectTemplates components, which are the ones that can be truly considered as tooling components, and in both cases you would want to always be in the latest version. These two are also not tightly coupled to the version of the Aspire.Hosting package that the AppHost project references, so they can be updated independently of the AppHost.
```

Comment:
The tight version coupling that DCP and Dashboard have makes sense in the explanation above. For Aspire.Hosting, I wasn't exactly clear. Is it just that it's not needed to load before restore so it doesn't need to be a workload so just include it in the set of packages?

Reply:
> Is it just that it's not needed to load before restore so it doesn't need to be a workload so just include it in the set of packages?

Correct. `Aspire.Hosting` is just a normal package in that sense.